File: GameCode/Player.java
package BunnyWarGame;

public class Player 
{
    private float healthBar;
    private float maxHealthBar; // Maximum health
    private float budget;
    private int weaponDamage;
    private int shieldProtection;
    private int levelsCompleted;

    public Player() 
    {
        this.maxHealthBar = 100; // Default maximum health
        this.healthBar = maxHealthBar; // Initialize current health to maximum
        this.budget = 0;
        this.weaponDamage = 15; // Default weapon damage
        this.shieldProtection = 0; // Default shield protection
        this.levelsCompleted = 0;
    }

    public float getHealthBar() 
    {
        return healthBar;
    }

    public void setHealthBar(float healthBar) 
    {
        this.healthBar = healthBar;
    }

    public float getMaxHealthBar() {
        return maxHealthBar;
    }

    public void setMaxHealthBar(float maxHealthBar) 
    {
        this.maxHealthBar = maxHealthBar;
    }

    public float getBudget() 
    {
        return budget;
    }

    public void setBudget(float budget) 
    {
        this.budget = budget;
    }

    public int getWeaponDamage() {
        return weaponDamage;
    }

    public void setWeaponDamage(int weaponDamage) 
    {
        this.weaponDamage = weaponDamage;
    }

    public int getShieldProtection() 
    {
        return shieldProtection;
    }

    public void setShieldProtection(int shieldProtection) 
    {
        this.shieldProtection = shieldProtection;
    }

    public int getLevelsCompleted() 
    {
        return levelsCompleted;
    }

    public void incrementLevelsCompleted() 
    {
        this.levelsCompleted++;
    }
}

File: bunnywarswebsite/Player.js
class Player {
    constructor() {
        this.maxHealthBar = 100;
        this.healthBar = this.maxHealthBar;
        this.budget = 0;
        this.weaponDamage = 15;
        this.shieldProtection = 0;
        this.levelsCompleted = 0;
    }

    getHealthBar() {
        return this.healthBar;
    }

    setHealthBar(healthBar) {
        this.healthBar = healthBar;
    }

    getMaxHealthBar() {
        return this.maxHealthBar;
    }

    setMaxHealthBar(maxHealthBar) {
        this.maxHealthBar = maxHealthBar;
    }

    getBudget() {
        return this.budget;
    }

    setBudget(budget) {
        this.budget = budget;
    }

    getWeaponDamage() {
        return this.weaponDamage;
    }

    setWeaponDamage(weaponDamage) {
        this.weaponDamage = weaponDamage;
    }

    getShieldProtection() {
        return this.shieldProtection;
    }

    setShieldProtection(shieldProtection) {
        this.shieldProtection = shieldProtection;
    }

    getLevelsCompleted() {
        return this.levelsCompleted;
    }

    incrementLevelsCompleted() {
        this.levelsCompleted++;
    }
}
